modules/auxiliary/admin/networking/cisco_dcnm_auth_bypass.rb

```diff
@@ -0,0 +1,178 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'securerandom'
+require 'base64'
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Cisco DCNM auth bypass',
+        'Description' => %q{
+          This exploit is able to add an admin account to a Cisco DCNM with credentials you can choose.
+          After that, you can login to the web interface with those credentials.
+          The only necessary condition is the more or less recent connection of an admin as this exploit
+          uses a kind of session stealing.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'MR_ME', # Amazing POC on www.exploit-db.com
+          'Yann Castel (yann.castel[at]orange.com)' # Metasploit module
+        ],
+        'References' =>
+          [
+            [ 'CVE', '2019-15975'],
+            [ 'EDB', '48018']
+          ],
+        'DisclosureDate' => '2020-06-01',
+        'DefaultOptions' => { 'SSL' => true }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(443),
+      OptInt.new('RETRIES', [true, 'Retry count for the attack', 50]),
+      OptString.new('TARGETURI', [true, 'The base path of the Cisco DCNM', '/']),
+      OptString.new('USERNAME', [true, 'The username of the admin account you want to add', random_string]),
+      OptString.new('PASSWORD', [true, 'The password of the admin account you want to add', random_string])
```

Comment:
We could further improve these randomization routines using the `Faker::Internet` module. There are routines for both [usernames](https://github.com/faker-ruby/faker/blob/master/lib/faker/default/internet.rb#L91) and [passwords](https://github.com/faker-ruby/faker/blob/master/lib/faker/default/internet.rb#L157).

For the usernames, it looks like they may not always be alpha numeric so you should check if there are any character limitations that may be problematic. The Faker option would generate a value that looks more legitimate and would be less likely to be an obvious indicator of compromise.

Reply:
Done !